Lesson2/SecondHomework.playground/Contents.swift

```diff
@@ -0,0 +1,50 @@
+/// An object, which is a list of elements with a type integer, organized according to the LIFO principle.
+class Stack {
+    fileprivate var array: [Int] = []
+    fileprivate var topElement: Int?
```

Comment:
For now this is more ok (this is not a mistake right now), but imagine the following:

```swift
let s = Stack()
s.topElement = 10
s.topElement
```

The last line returns `10` despite the stack is empty. The reason is the violation of encapsulation. Could you please try to improve the implementation? (This is NOT a mandatory task).

Reply:
@dersim-davaod, Will it be right if I do it like this?

 fileprivate var array: [Int] = [] {
        didSet {
            topElement = array.last
        }
    }
    private var topElement: Int?

Reply:
Then you might still have an issue with encapsulation:
```swift
let s = Stack()
s.array.append(10)
```

You should be able to change the state of the object via methods only, not directly via fields. Try to avoid using `fileprivate` at all (Imagine your subclass belongs to another file).

P.S.: You can use backticks to insert your code:
\```
```
your code here
```
\```

or

\```swift
```swift
swift code here
```
\```

Reply:
I changed the logic so as not to violate the principles of encapsulation